src/Features/Core/Diagnostics/EngineV1/DiagnosticAnalyzerDriver.cs

```diff
@@ -343,14 +376,16 @@ private IEnumerable<Diagnostic> GetFilteredDocumentDiagnosticsCore(IEnumerable<D
                     }
                     catch (Exception e) when (CatchAnalyzerException(e, analyzer))
                     {
-                        var exceptionDiagnostics = AnalyzerExceptionToDiagnostics(analyzer, e, _cancellationToken);
-                        return model == null ? exceptionDiagnostics : GetFilteredDocumentDiagnostics(exceptionDiagnostics, model.Compilation);
+                        var exceptionDiagnostics = AnalyzerExceptionToDiagnostic(analyzer, e, _cancellationToken);
+                        ReportAnalyzerExceptionDiagnostic(analyzer, exceptionDiagnostics, compilation);
```

Comment:
Needs a null check for value returned from AnalyzerExceptionToDiagnostic. I'll also fix the redundant spaces added in various files in this PR.